external-testing/README.md

```diff
@@ -0,0 +1,65 @@
+# External Testing
+
+Let's talk about testing process against external resources. What does that
+mean? Well, talking to GitHub API, COPR API, running `fedpkg`, cloning
+repositories...
+
+
+## Use cases
+
+* Interact with an external API (such as GitHub, COPR, Pagure, FAS, k8s)
+  * Authenticated vs. unauthenticated calls.
+* Clone a git repository.
+* Construct a git repository.
+* Run a command to perform an action.
+  * Changes happening only locally.
+  * RO access to a remote service.
+  * R/W access to a remote service.
+* Combination of multiple use cases into a single test case.
+
+
+## Solutions
+
+* flexmock
+
+* pytest fixtures
+
+* requre
+
+* VCRpy
+
+* Betamax
+
+
+## VCRpy vs. requre
+
+| A feature | VCRpy | requre |
+|-----------|-------|--------|
+| Record & replay for python requests | ✓ | ✓ |
+| Match on call stack | ✘ | ✓ |
+| Match on URL & request type | ✓ | ✘ |
+| Calling the same URL twice works as expected | ✓ | ✘ |
+| Record & replay for arbitrary functions | ✘ | ✓ |
+| Filtering sensitive data out of recorded responses | ✓ [1] | ✓ [2] |
+| Easy to use | ✓ [3] | ✘ [4] |
+| Actively maintained | ✘ | ✓ |
+
+* [1] https://vcrpy.readthedocs.io/en/latest/advanced.html#filter-sensitive-data-from-the-request
+* [2] https://requre.readthedocs.io/en/latest/usages/postprocessing.html
+* [3] [You can use a decorator or a context manager](https://vcrpy.readthedocs.io/en/latest/usage.html)
+* [4] [Tenths of lines of code](https://github.com/packit-service/packit/blob/master/tests_recording/__init__.py#L16) [which are very hard to understand](https://github.com/packit-service/packit/blob/master/tests_recording/testbase.py#L22)
```

Comment:
these tenth of lines are caused to be able to work more generic, in case you would like to record just requests module and nothing else:
 ```
upgraded_import_system = (upgrade_import_system().decorate(where="^requests$", what="Session.send", decorator=RequestResponseHandling.decorator(item_list=[])))
```
exactly stores all requests to `request` module.

Reply:
sacrificing user experience over configurability

Reply:
This is a solvable problem. Things that will don't take much time and will help a lot:
- renaming of the functions and  arguments
- fixtures for common replacements (e.g. requests)
- add some basic examples to docs